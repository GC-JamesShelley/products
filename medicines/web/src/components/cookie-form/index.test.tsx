import { shallow } from 'enzyme';
import React from 'react';
import CookieForm from './index';

describe(CookieForm, () => {
  it('should render 🍪', () => {
    // tslint:disable-next-line: no-empty
    const noop = () => {};
    const component = shallow(
      <CookieForm storageAllowed setStorageAllowed={noop} />,
    );
    expect(component).toMatchSnapshot();
  });

  it('should render 🚫', () => {
    // tslint:disable-next-line: no-empty
    const noop = () => {};
    const component = shallow(
      <CookieForm storageAllowed={false} setStorageAllowed={noop} />,
    );
    expect(component).toMatchSnapshot();
  });
});
