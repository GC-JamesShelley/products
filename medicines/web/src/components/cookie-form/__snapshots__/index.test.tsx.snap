// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookieForm should render 🍪 1`] = `
<styled.section>
  <styled.h3>
    Toggle cookie permissions
  </styled.h3>
  <form
    onSubmit={[Function]}
  >
    <p>
      <label
        htmlFor="cookie-on"
      >
        <input
          aria-checked={true}
          checked={true}
          id="cookie-on"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="on"
        />
        On
      </label>
      <label
        htmlFor="cookie-off"
      >
        <input
          aria-checked={true}
          checked={false}
          id="cookie-off"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="off"
        />
        Off
      </label>
    </p>
    <styled.div
      dangerouslySetInnerHTML={
        Object {
          "__html": Object {
            "process": [Function],
          },
        }
      }
    />
    <p>
      <button>
        Save your preferences
      </button>
    </p>
  </form>
</styled.section>
`;

exports[`CookieForm should render 🚫 1`] = `
<styled.section>
  <styled.h3>
    Toggle cookie permissions
  </styled.h3>
  <form
    onSubmit={[Function]}
  >
    <p>
      <label
        htmlFor="cookie-on"
      >
        <input
          aria-checked={false}
          checked={false}
          id="cookie-on"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="on"
        />
        On
      </label>
      <label
        htmlFor="cookie-off"
      >
        <input
          aria-checked={false}
          checked={true}
          id="cookie-off"
          name="cookie"
          onChange={[Function]}
          type="radio"
          value="off"
        />
        Off
      </label>
    </p>
    <styled.div
      dangerouslySetInnerHTML={
        Object {
          "__html": Object {
            "process": [Function],
          },
        }
      }
    />
    <p>
      <button>
        Save your preferences
      </button>
    </p>
  </form>
</styled.section>
`;
